src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -2068,6 +2148,25 @@ def set_frame_2d_labels(
     # map from item's data column label to its position in the joined frame or item_column_values
     item_data_col_label_to_pos_map: dict[Hashable, int] = {}
     if item_is_frame:
+        from snowflake.snowpark.modin.plugin.compiler.snowflake_query_compiler import (
+            SnowflakeQueryCompiler,
+        )
+
+        # If `item` is from a Series (rather than a Dataframe), flip the series item values to apply them
+        # across columns rather than rows.
+        is_multi_col_set = (
+            (isinstance(columns, Sized) and len(columns) > 1)
+            or isinstance(columns, slice)
```

Comment:
slice and qc case can be single column right?

Reply:
Good catch, thank you!